test/AutoDiff/generics.swift

```diff
@@ -57,25 +57,75 @@ struct SupervisedTrainer<Model : Layer> {
 }
 
 // Tests TF-440.
-struct TF_440_Input<Input: Differentiable, State: Differentiable>: Differentiable {
-    var input: Input
-    var state: State
+struct TF_440_Input<Input: Differentiable, State: Differentiable>
+  : Differentiable {
+  var input: Input
+  var state: State
 }
 struct TF_440<T : Differentiable> {
-    @differentiable
-    func applied(to input: TF_440_Input<Float, Float>) -> Float {
-        return input.state
-    }
-
-    @differentiable
-    func applied(to input: TF_440_Input<T, Float>) -> Float {
-        return input.state
-    }
-
-    @differentiable
-    func applied(to input: TF_440_Input<T, Float>) -> T {
-        return input.input
-    }
+  @differentiable
+  func applied(to input: TF_440_Input<Float, Float>) -> Float {
+    return input.state
+  }
+
+  @differentiable
+  func applied(to input: TF_440_Input<T, Float>) -> Float {
+    return input.state
+  }
+
+  @differentiable
+  func applied(to input: TF_440_Input<T, Float>) -> T {
+    return input.input
+  }
+}
+
+// Tests TF-508 (Hack)
```

Comment:
You can drop `(Hack)` from here since the test is not a hack (the implementation is).

Reply:
Oh ya true! I just wanted to differentiate it for when this bug is fully fixed, but ya the test doesn't matter.